### PR TITLE
JSON Schema Export test enhancements

### PR DESCRIPTION
**Pull request by jgibson**
This branch contains new tests to support the JSON Schema exporter. For the most part it is backwards compatible. However there are a handful of tests (Choice, ChoiceOfChoice, Group, and GroupWithChoiceOfChoice) that are not because they were testing no-longer legal things like choices on fields.

The only other change is that if there is an AssertionError while loading a test fixture then it is handled as a test failure rather than as a skipped test. This may make assertion-error a good candidate for a peer dependency.

### CODE REVIEW

File: export.js [hunk 1]
```diff
@@ -15,6 +16,9 @@ function commonExportTests(exportFn, expectedFn, expectedErrorsFn) {
         errors: expectedErrorsFn(name)
       };
     } catch (e) {
+      if (e instanceof AssertionError) {
```
Comment (cmoesel) [inline export.js]: Looking at the `AssertionError` source code [here](https://github.com/chaijs/assertion-error/blob/master/index.js#L81-L91), I think we can avoid pulling in `AssertionError` as a dependency at all by doing this:

```js
if (e instanceof Error && e.name === 'AssertionError')
```

This way we could avoid needing to declare it as a peer dependency (or needing to declare it at all).

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "shr-test-helpers",
-  "version": "5.0.3",
+  "version": "5.1.0-beta.2",
```
Comment (cmoesel) [inline package.json]: Is this still a beta?

**Review by cmoesel — CHANGES_REQUESTED** [review]
Let me know what you think regarding my suggestion about `AssertionError`.

File: export.js (hunk 1 above)
Comment (jgibson) [reply export.js]: That's true. I think that the only way things could go wrong is if an exporter threw a different `AssertionError` as part of loading a fixture and did not intend it to be a test failure. That seems pretty unlikely.

The only other advantage of declaring it as a peer dependency is that we'd get to experiment with how they behave before mucking about with peer deps for all of shr-models.

Comment (cmoesel) [reply export.js]: I'm not too concerned about some other different `AssertionError`.  I think we'd want the error to pop up no matter what anyway.

As for trying out peer dependencies... nah.  I think that when we have a chance to try them w/ shr-models, we should be able to fully test them OK without needing a pre-experiment.

Does that sound OK to you?

**Review by cmoesel — APPROVED** [review]

**Pull request merged**